--- package/chinadns-ng/Makefile	2022-07-29 01:18:54.714266988 +0800
+++ package/chinadns-ng/Makefile	2022-07-29 01:21:00.144266941 +0800
@@ -38,6 +38,7 @@ define Package/chinadns-ng/conffiles
 /etc/chinadns-ng/chnroute6.txt
 /etc/chinadns-ng/gfwlist.txt
 /etc/chinadns-ng/chinalist.txt
+/etc/chinadns-ng/dnsmasq-chinadns.conf
 endef
 
 define Package/chinadns-ng/install
@@ -52,6 +53,7 @@ define Package/chinadns-ng/install
 	$(INSTALL_DATA) files/chnroute6.txt $(1)/etc/chinadns-ng
 	$(INSTALL_DATA) files/gfwlist.txt $(1)/etc/chinadns-ng
 	$(INSTALL_DATA) files/chinalist.txt $(1)/etc/chinadns-ng
+	$(INSTALL_DATA) files/dnsmasq-chinadns.conf $(1)/etc/chinadns-ng
 endef
 
 define Package/chinadns-ng/postrm
