--- feeds/packages/net/dnsforwarder/files/etc/init.d/dnsforwarder	2022-09-01 08:40:52.649956481 +0800
+++ feeds/packages/net/dnsforwarder/files/etc/init.d/dnsforwarder	2022-09-01 08:41:59.929956455 +0800
@@ -12,7 +12,7 @@ DNSFORWARDER_CONF=/tmp/dnsforwarder.conf
 add_cron()
 {
 	sed -i '/dnsforwarder/d' $CRON_FILE
-	echo '*/5 * * * * /etc/init.d/dnsforwarder health' >> $CRON_FILE
+	#echo '*/5 * * * * /etc/init.d/dnsforwarder health' >> $CRON_FILE
 	echo '0 1 * * 0 /etc/init.d/dnsforwarder makegfwlist' >> $CRON_FILE
 	crontab $CRON_FILE
 }
