--- target/linux/ramips/dts/mt7621.dtsi	2022-08-31 23:49:48.820003170 +0800
+++ target/linux/ramips/dts/mt7621.dtsi	2022-08-31 23:53:12.850003092 +0800
@@ -557,8 +557,8 @@
 
 	hnat: hnat@1e100000 {
 		compatible = "mediatek,mtk-hnat_v1";
-		ext-devices = "apcli0", "apclii0","apclix0",
-			"ra0", "rai0", "rax0", "wlan0", "wlan1";
+		/* ext-devices = "apcli0", "apclii0","apclix0",
+			"ra0", "rai0", "rax0", "wlan0", "wlan1"; */
 		reg = <0x1e100000 0x3000>;
 
 		resets = <&ethsys 0>;
