--- package/lean/default-settings/files/zzz-default-settings	2022-08-22 13:27:55.604529727 +0800
+++ package/lean/default-settings/files/zzz-default-settings	2022-08-22 13:28:56.904529704 +0800
@@ -37,6 +37,9 @@ sed -i "s/# //g" /etc/opkg/distfeeds.con
 sed -i '/openwrt_luci/ { s/snapshots/releases\/18.06.9/g; }'  /etc/opkg/distfeeds.conf
 
 sed -i '/REDIRECT --to-ports 53/d' /etc/firewall.user
+sed -i '/53 -j RETURN -d 192.168.0.0\/16/d' /etc/firewall.user
+echo 'iptables -t nat -A PREROUTING -p udp --dport 53 -j RETURN -d 192.168.0.0/16' >> /etc/firewall.user
+echo 'iptables -t nat -A PREROUTING -p tcp --dport 53 -j RETURN -d 192.168.0.0/16' >> /etc/firewall.user
 echo 'iptables -t nat -A PREROUTING -p udp --dport 53 -j REDIRECT --to-ports 53' >> /etc/firewall.user
 echo 'iptables -t nat -A PREROUTING -p tcp --dport 53 -j REDIRECT --to-ports 53' >> /etc/firewall.user
 echo '[ -n "$(command -v ip6tables)" ] && ip6tables -t nat -A PREROUTING -p udp --dport 53 -j REDIRECT --to-ports 53' >> /etc/firewall.user
