--- feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-rules	2022-08-30 21:41:27.743478022 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-rules	2022-08-30 21:42:46.003477993 +0800
@@ -214,6 +214,14 @@ ac_rule() {
        return $?
 }

+iptables_config() {
+       source /etc/ssrplus/iptables_config.sh 2>/dev/null || {
+               loger 3 "Can't config iptables, please check the config."
+               exit 1
+       }
+       return $?
+}
+
 tp_rule() {
        [ -n "$TPROXY" ] || return 0
        ip rule add fwmark 0x01/0x01 table 100
@@ -420,7 +428,7 @@ case "$TPROXY" in
        ;;
 esac

-flush_r && fw_rule && ipset_r && ac_rule && tp_rule && gen_include
+flush_r && fw_rule && ipset_r && ac_rule && tp_rule && iptables_config && gen_include
 RET=$?
 [ "$RET" = 0 ] || loger 3 "Start failed!"
 exit $RET

ssrplus添加53端口绕过是因为AC在53劫持规则之前，必须绕过才可走到劫持那步；
仅黑名单UDP代理并不代表全局代理，只作为UDP代理分流规则条件中的一个，比如黑名单中包含中国域名，那就不会代理
--- /dev/null	1970-01-17 13:33:51.799999999 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/etc/ssrplus/iptables_config.sh	2022-08-31 15:24:13.390562124 +0800
@@ -0,0 +1,7 @@
+#$IPT -A SS_SPEC_WAN_FW -p tcp -m multiport --dport 85,86 -j REDIRECT --to-ports $local_port
+#$IPT -I SS_SPEC_WAN_AC 1 -p tcp --dport 443 -j RETURN -d 1.1.1.1
+#$IPT -I SS_SPEC_WAN_AC 2 -p tcp --dport 443 -j RETURN -d 8.8.8.8
+#$IPT -I SS_SPEC_WAN_AC 1 -i br-lan -p udp --dport 53 -j RETURN
+#$IPT -I SS_SPEC_WAN_AC 2 -i br-lan -p tcp --dport 53 -j RETURN
+#$IPT -I SS_SPEC_WAN_AC 1 -p udp -m multiport --dport 80,443 -m set ! --match-set china dst -j RETURN
+#iptables -t mangle -I SS_SPEC_TPROXY 1 -p udp -m set ! --match-set blacklist dst -j RETURN || true
