--- package/base-files/files/etc/rc.local	2022-07-27 17:37:25.852250698 +0800
+++ package/base-files/files/etc/rc.local	2022-07-27 17:38:29.942250674 +0800
@@ -1,4 +1,13 @@
 # Put your custom commands here that should be executed once
 # the system init finished. By default this file does nothing.
+echo \
+'UDPLocal 127.0.0.1:534,[::1]:534
+TCPLocal 127.0.0.1:534,[::1]:534
+UDPGroup 223.5.5.5 * on
+EnableTCPtoUDP true
+UseCache true
+MemoryCache true' > /tmp/dnsforwarder-bplan.conf
+ln -s /usr/bin/dnsforwarder /tmp/dnsforwarder-bplan
+/tmp/dnsforwarder-bplan -d -f /tmp/dnsforwarder-bplan.conf
 
 exit 0
