--- package/base-files/files/etc/rc.local	2022-07-27 17:37:25.852250698 +0800
+++ package/base-files/files/etc/rc.local	2022-07-27 17:38:29.942250674 +0800
@@ -1,4 +1,9 @@
 # Put your custom commands here that should be executed once
 # the system init finished. By default this file does nothing.
+
+#bropc
+/usr/share/dnsforwarder-bropc/genlist.sh custom
+ln -s /usr/bin/dnsforwarder /tmp/dnsforwarder-bropc
+/tmp/dnsforwarder-bropc -d -f /etc/dnsforwarder-bropc/dnsforwarder.config
 
 exit 0
