--- feeds/packages/net/dnsforwarder/Makefile	2022-08-30 13:22:14.525448687 +0800
+++ feeds/packages/net/dnsforwarder/Makefile	2022-08-30 13:30:15.625448503 +0800
@@ -9,14 +9,14 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=dnsforwarder
 PKG_VERSION:=6.1.15
 PKG_RELEASE:=11
 
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_URL:=https://github.com/1715173329/dnsforwarder.git
+PKG_SOURCE_URL:=https://github.com/lifenjoiner/dnsforwarder.git
 
 PKG_LICENSE:=GPL-3.0
 PKG_LICENSE_FILE:=LICENSE
