--- feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-rules	2022-08-30 21:41:27.743478022 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-rules	2022-08-30 21:42:46.003477993 +0800
@@ -214,6 +214,14 @@ ac_rule() {
        return $?
 }

+iptables_config() {
+       source /etc/ssrplus/iptables_config.sh 2>/dev/null || {
+               loger 3 "Can't config iptables, please check the config."
+               exit 1
+       }
+       return $?
+}
+
 tp_rule() {
        [ -n "$TPROXY" ] || return 0
        ip rule add fwmark 0x01/0x01 table 100
@@ -420,7 +428,7 @@ case "$TPROXY" in
        ;;
 esac

-flush_r && fw_rule && ipset_r && ac_rule && tp_rule && gen_include
+flush_r && fw_rule && ipset_r && ac_rule && tp_rule && iptables_config && gen_include
 RET=$?
 [ "$RET" = 0 ] || loger 3 "Start failed!"
 exit $RET
