--- feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-rules	2022-08-30 21:41:27.743478022 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-rules	2022-08-30 21:42:46.003477993 +0800
@@ -63,8 +63,21 @@ flush_r() {
 			$ipt -F ${chain:1} 2>/dev/null && $ipt -X ${chain:1}
 		done
 	}
+	flush_ip6tables() {
+		local ipt="ip6tables -t $1"
+		local DAT=$(ip6tables-save -t $1)
+		eval $(echo "$DAT" | grep "$TAG" | sed -e 's/^-A/$ipt -D/' -e 's/$/;/')
+		for chain in $(echo "$DAT" | awk '/^:SS_SPEC/{print $1}'); do
+			$ipt -F ${chain:1} 2>/dev/null && $ipt -X ${chain:1}
+		done
+	}
 	flush_iptables nat
 	flush_iptables mangle
+	flush_iptables filter
+	if [ -n "$(command -v ip6tables)" ]; then
+		flush_ip6tables nat
+		flush_ip6tables filter
+	fi
 	ip rule del fwmark 0x01/0x01 table 100 2>/dev/null
 	ip route del local 0.0.0.0/0 dev lo table 100 2>/dev/null
 	ipset -X ss_spec_lan_ac 2>/dev/null
@@ -214,6 +227,25 @@ ac_rule() {
 	return $?
 }
 
+iptables_config() {
+	iptables -t filter -N SS_SPEC_CUS_LAN_FWD
+	iptables -t filter -I zone_lan_forward 1 -m comment --comment _SS_SPEC_RULE_ -j SS_SPEC_CUS_LAN_FWD
+	iptables -t nat -N SS_SPEC_CUS_WAN_AC
+	iptables -t nat -I PREROUTING 2 -m comment --comment _SS_SPEC_RULE_ -j SS_SPEC_CUS_WAN_AC
+	if [ -n "$(command -v ip6tables)" ]; then
+		ip6tables -t filter -N SS_SPEC_CUS_LAN_FWD
+		ip6tables -t filter -I zone_lan_forward 1 -m comment --comment _SS_SPEC_RULE_ -j SS_SPEC_CUS_LAN_FWD
+		ip6tables -t nat -N SS_SPEC_CUS_WAN_AC
+		ip6tables -t nat -I PREROUTING 1 -m comment --comment _SS_SPEC_RULE_ -j SS_SPEC_CUS_WAN_AC
+	fi
+
+	source /etc/ssrplus/iptables_config.sh || {
+		loger 3 "Can't config iptables, please check the config."
+		exit 1
+	}
+	return $?
+}
+
 tp_rule() {
 	[ -n "$TPROXY" ] || return 0
 	ip rule add fwmark 0x01/0x01 table 100
@@ -304,7 +336,13 @@ gen_include() {
 	[ -n "$FWI" ] || return 0
 	extract_rules() {
 		echo "*$1"
-		iptables-save -t $1 | grep SS_SPEC_ | sed -e "s/^-A \(OUTPUT\|PREROUTING\)/-I \1 1/"
+		iptables-save -t $1 | grep SS_SPEC_ | sed '/SS_SPEC_CUS_/d' | sed -e "s/^-A \(OUTPUT\|PREROUTING\)/-I \1 1/"
+		iptables-save -t $1 | grep SS_SPEC_CUS_ | sed -e "s/^-A \(zone_lan_forward\)/-I \1 1/" | sed -e "s/^-A \(PREROUTING\)/-I \1 2/"
+		echo 'COMMIT'
+	}
+	extract_rules6() {
+		echo "*$1"
+		ip6tables-save -t $1 | grep SS_SPEC_CUS_ | sed -e "s/^-A \(PREROUTING\|zone_lan_forward\)/-I \1 1/"
 		echo 'COMMIT'
 	}
 	cat <<-EOF >>$FWI
@@ -312,6 +350,13 @@ gen_include() {
 		iptables-restore -n <<-EOT
 		$(extract_rules nat)
 		$(extract_rules mangle)
+		$(extract_rules filter)
+		EOT
+
+		ip6tables-save -c | grep -v "SS_SPEC" | ip6tables-restore -c
+		ip6tables-restore -n <<-EOT
+		$(extract_rules6 nat)
+		$(extract_rules6 filter)
 		EOT
 	EOF
 	return 0
@@ -420,7 +465,7 @@ case "$TPROXY" in
 	;;
 esac
 
-flush_r && fw_rule && ipset_r && ac_rule && tp_rule && gen_include
+flush_r && fw_rule && ipset_r && ac_rule && tp_rule && iptables_config && gen_include
 RET=$?
 [ "$RET" = 0 ] || loger 3 "Start failed!"
 exit $RET
