--- feeds/helloworld/luci-app-ssr-plus/root/usr/share/shadowsocksr/update.lua	2022-10-02 12:16:54.528020038 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/share/shadowsocksr/update.lua	2022-10-02 12:17:30.008020024 +0800
@@ -17,8 +17,8 @@ local ip_pattern = "^%d+%.%d+%.%d+%.%d+"
 local domain_pattern = "([%w%-%_]+%.[%w%.%-%_]+)[%/%*]*"
 local excluded_domain = {"apple.com", "sina.cn", "sina.com.cn", "baidu.com", "byr.cn", "jlike.com", "weibo.com", "zhongsou.com", "youdao.com", "sogou.com", "so.com", "soso.com", "aliyun.com", "taobao.com", "jd.com", "qq.com"}
 -- gfwlist parameter
-local mydnsip = '127.0.0.1'
-local mydnsport = '5335'
+local mydnsip = uci:get_first("shadowsocksr", "global", "mydnsip")
+local mydnsport = uci:get_first("shadowsocksr", "global", "mydnsport")
 local ipsetname = 'gfwlist'
 local bc = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/'
 -- base64decoding

--- feeds/helloworld/luci-app-ssr-plus/root/usr/share/shadowsocksr/gfw2ipset.sh	2022-10-02 17:40:21.315292865 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/share/shadowsocksr/gfw2ipset.sh	2022-10-02 19:54:13.599505274 +0800
@@ -8,6 +8,13 @@ netflix() {
 	cat /etc/ssrplus/netflix.list | sed '/^$/d' | sed '/#/d' | sed "/.*/s/.*/server=\/&\/127.0.0.1#$1\nipset=\/&\/netflix/" >$TMP_DNSMASQ_PATH/netflix_forward.conf
 }
 mkdir -p $TMP_DNSMASQ_PATH
+for file in /etc/ssrplus/oversea_list.conf /etc/ssrplus/gfw_list.conf /etc/ssrplus/gfw_base.conf; do
+	dns=$(grep '^server=/' -m1 $file)
+	dns=${dns##*/}
+	if [ $dns != "$dns_ip#$dns_port" ]; then
+		sed -i "s/$dns/$dns_ip#$dns_port/g" $file
+	fi
+done
 if [ "$(uci_get_by_type global run_mode router)" == "oversea" ]; then
 	cp -rf /etc/ssrplus/oversea_list.conf $TMP_DNSMASQ_PATH/
 else
@@ -31,7 +35,7 @@ for line in $(cat /etc/ssrplus/white.lis
 for line in $(cat /etc/ssrplus/white.list); do sed -i "/$line/d" $TMP_DNSMASQ_PATH/gfw_base.conf; done
 for line in $(cat /etc/ssrplus/deny.list); do sed -i "/$line/d" $TMP_DNSMASQ_PATH/gfw_list.conf; done
 for line in $(cat /etc/ssrplus/deny.list); do sed -i "/$line/d" $TMP_DNSMASQ_PATH/gfw_base.conf; done
-cat /etc/ssrplus/black.list | sed '/^$/d' | sed '/#/d' | sed "/.*/s/.*/server=\/&\/127.0.0.1#$dns_port\nipset=\/&\/blacklist/" >$TMP_DNSMASQ_PATH/blacklist_forward.conf
+cat /etc/ssrplus/black.list | sed '/^$/d' | sed '/#/d' | sed "/.*/s/.*/server=\/&\/$dns_ip#$dns_port\nipset=\/&\/blacklist/" >$TMP_DNSMASQ_PATH/blacklist_forward.conf
 cat /etc/ssrplus/white.list | sed '/^$/d' | sed '/#/d' | sed "/.*/s/.*/server=\/&\/127.0.0.1\nipset=\/&\/whitelist/" >$TMP_DNSMASQ_PATH/whitelist_forward.conf
 cat /etc/ssrplus/deny.list | sed '/^$/d' | sed '/#/d' | sed "/.*/s/.*/address=\/&\//" >$TMP_DNSMASQ_PATH/denylist.conf
 if [ "$(uci_get_by_type global adblock 0)" == "1" ]; then

--- feeds/helloworld/luci-app-ssr-plus/root/etc/init.d/shadowsocksr	2022-10-02 17:33:30.035293022 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/etc/init.d/shadowsocksr	2022-10-02 20:03:12.869505068 +0800
@@ -25,7 +25,8 @@ local_config_file=
 shunt_dns_config_file=
 tmp_local_port=
 ARG_UDP=
-dns_port="5335"            #dns port
+dns_ip=$(uci get shadowsocksr.@global[0].mydnsip)
+dns_port=$(uci get shadowsocksr.@global[0].mydnsport)            #dns port
 china_dns_port="5333"      #china_dns_port
 tmp_dns_port="300"         #dns2socks temporary port
 tmp_udp_port="301"         #udp temporary port
@@ -186,8 +187,10 @@ start_dns() {
 		esac
 		case "$ssrplus_dns" in
 		1)
-			ln_start_bin $(first_type dns2tcp) dns2tcp -L 127.0.0.1#$dns_port -R ${dnsserver/:/#}
-			pdnsd_enable_flag=1
+			if [[ "$dns_ip" == "127.0.0.1" ]] && [[ "$dns_port" == "5335" ]]; then
+				ln_start_bin $(first_type dns2tcp) dns2tcp -L 127.0.0.1#$dns_port -R ${dnsserver/:/#}
+				pdnsd_enable_flag=1
+			fi
 			;;
 		2)
 			ln_start_bin $(first_type microsocks) microsocks -i 127.0.0.1 -p $tmp_dns_port ssrplus-dns
@@ -205,7 +208,7 @@ start_dns() {
 				"wan_114") chinadns="$wandns,114.114.114.114" ;;
 				esac
 
-				ln_start_bin $(first_type chinadns-ng) chinadns-ng -l $china_dns_port -4 china -p 3 -c ${chinadns/:/#} -t 127.0.0.1#$dns_port -N -f -r
+				ln_start_bin $(first_type chinadns-ng) chinadns-ng -l $china_dns_port -4 china -p 3 -c ${chinadns/:/#} -t $dns_ip#$dns_port -N -f -r
 
 				cat <<-EOF >> "$TMP_DNSMASQ_PATH/chinadns_fixed_server.conf"
 					no-poll
@@ -986,6 +989,8 @@ reset() {
 		set shadowsocksr.@global[0].chnroute_url='https://ispip.clang.cn/all_cn.txt'
 		set shadowsocksr.@global[0].nfip_url='https://fastly.jsdelivr.net/gh/QiuSimons/Netflix_IP/NF_only.txt'
 		set shadowsocksr.@global[0].adblock_url='https://anti-ad.net/anti-ad-for-dnsmasq.conf'
+		set shadowsocksr.@global[0].mydnsip='127.0.0.1'
+		set shadowsocksr.@global[0].mydnsport='5335'
 		add shadowsocksr server_subscribe
 		set shadowsocksr.@server_subscribe[0].proxy='0'
 		set shadowsocksr.@server_subscribe[0].auto_update_time='2'

--- feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-monitor	2022-12-21 23:39:14.612940239 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-monitor	2022-12-21 23:48:55.192940017 +0800
@@ -118,7 +118,7 @@ while [ "1" == "1" ]; do #死循环
 			""|"wan_114") chinadns="$wandns,114.114.114.114" ;;
 			esac
 			kill -9 $(busybox ps -w | grep $TMP_BIN_PATH/chinadns-ng | grep -v grep | awk '{print $1}') >/dev/null 2>&1
-			ln_start_bin $(first_type chinadns-ng) chinadns-ng -l $china_dns_port -4 china -p 3 -c ${chinadns/:/#} -t 127.0.0.1#$dns_port -N -f -r
+			ln_start_bin $(first_type chinadns-ng) chinadns-ng -l $china_dns_port -4 china -p 3 -c ${chinadns/:/#} -t $dns_ip#$dns_port -N -f -r
 		fi
 	fi
 done
