--- feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-monitor	2022-09-01 20:29:49.646396524 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-monitor	2022-09-01 20:29:25.976396533 +0800
@@ -109,4 +109,8 @@ while [ "1" == "1" ]; do #死循环
 			ln_start_bin $(first_type dns2socks) dns2socks 127.0.0.1:$tmp_dns_port $dnsserver:$dnsport 127.0.0.1:$dns_port -q
 		fi
 	fi
+	#dnsforwarder
+	if [ $(uci get dnsforwarder.@arguments[0].enabled 2>/dev/null | echo 0) == 1 ]; then
+		/etc/init.d/dnsforwarder health #如果进程挂掉就重启它
+	fi
 done

--- feeds/helloworld/luci-app-ssr-plus/root/etc/init.d/shadowsocksr	2022-09-01 20:28:44.376396549 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/etc/init.d/shadowsocksr	2022-09-01 20:28:28.516396555 +0800
@@ -187,6 +187,9 @@ start_dns() {
 			;;
 		esac
 	fi
+	if [ $(uci get dnsforwarder.@arguments[0].enabled 2>/dev/null | echo 0) == 1 ]; then #重启ssrplus会使dnsforwarder退出故重启
+		/etc/init.d/dnsforwarder health #如果进程挂掉就重启它
+	fi
 }
 
 gen_service_file() {
