--- feeds/helloworld/luci-app-ssr-plus/root/etc/init.d/shadowsocksr	2022-12-18 14:22:16.896790134 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/etc/init.d/shadowsocksr	2022-12-18 14:22:49.276790122 +0800
@@ -210,7 +210,7 @@ start_dns() {
 				"wan_114") chinadns="$wandns,$chinadns_second" ;;
 				esac
 
-				ln_start_bin $(first_type chinadns-ng) chinadns-ng -l $china_dns_port -4 china -6 china6 -p 3 -c ${chinadns/:/#} -t $dns_ip#$dns_port -N=gt -f -r -M -g /etc/ssrplus/chinadns_black.list -m /etc/ssrplus/chinadns_white.list
+				ln_start_bin $(first_type chinadns-ng) chinadns-ng -l $china_dns_port -4 china -6 china6 -p 3 -c ${chinadns/:/#} -t $dns_ip#$dns_port -N=gt -f -r -M -n -g /etc/ssrplus/chinadns_black.list -m /etc/ssrplus/chinadns_white.list
 
 				cat <<-EOF >> "$TMP_DNSMASQ_PATH/chinadns_fixed_server.conf"
 					no-poll

--- feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-monitor	2022-12-22 00:07:41.432939588 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-monitor	2022-12-22 00:11:11.922939507 +0800
@@ -119,7 +119,7 @@ while [ "1" == "1" ]; do #死循环
 			""|"wan_114") chinadns="$wandns,$chinadns_second" ;;
 			esac
 			kill -9 $(busybox ps -w | grep $TMP_BIN_PATH/chinadns-ng | grep -v grep | awk '{print $1}') >/dev/null 2>&1
-			ln_start_bin $(first_type chinadns-ng) chinadns-ng -l $china_dns_port -4 china -6 china6 -p 3 -c ${chinadns/:/#} -t $dns_ip#$dns_port -N=gt -f -r -M -g /etc/ssrplus/chinadns_black.list -m /etc/ssrplus/chinadns_white.list
+			ln_start_bin $(first_type chinadns-ng) chinadns-ng -l $china_dns_port -4 china -6 china6 -p 3 -c ${chinadns/:/#} -t $dns_ip#$dns_port -N=gt -f -r -M -n -g /etc/ssrplus/chinadns_black.list -m /etc/ssrplus/chinadns_white.list
 		fi
 	fi
 done
