--- feeds/helloworld/luci-app-ssr-plus/root/etc/init.d/shadowsocksr	2023-02-19 11:40:17.300236408 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/etc/init.d/shadowsocksr	2023-02-19 11:43:08.210236343 +0800
@@ -898,6 +898,7 @@ start() {
 	mkdir -p /var/run /var/lock /var/log /tmp/dnsmasq.d $TMP_BIN_PATH $TMP_DNSMASQ_PATH
 	echo "conf-dir=${TMP_DNSMASQ_PATH}" >"/tmp/dnsmasq.d/dnsmasq-ssrplus.conf"
 	if load_config; then
+		/usr/share/shadowsocksr/when_start.sh
 		Start_Run
 		start_rules
 		start_dns
@@ -965,6 +966,7 @@ stop() {
 	fi
 	$PS -w | grep -v "grep" | grep ssr-monitor | awk '{print $1}' | xargs kill -9 >/dev/null 2>&1 &
 	$PS -w | grep -v "grep" | grep "sleep 0000" | awk '{print $1}' | xargs kill -9 >/dev/null 2>&1 &
+	/usr/share/shadowsocksr/when_stop.sh
 	( \
 		# Graceful kill first, so programs have the chance to stop its subprocesses
 		$PS -w | grep -v "grep" | grep "$TMP_PATH" | awk '{print $1}' | xargs kill >/dev/null 2>&1 ; \
