--- feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-rules	2023-02-20 22:34:15.155025950 +0800
+++ feeds/helloworld/luci-app-ssr-plus/root/usr/bin/ssr-rules	2023-02-20 22:33:56.935025957 +0800
@@ -323,7 +323,13 @@ gen_include() {
 	[ -n "$FWI" ] || return 0
 	extract_rules() {
 		echo "*$1"
-		iptables-save -t $1 | grep SS_SPEC_ | sed -e "s/^-A \(OUTPUT\|PREROUTING\)/-I \1 1/"
+		iptables-save -t $1 | grep 'SS_SPEC_[^CUS_]' | sed -e "s/^-A \(OUTPUT\|PREROUTING\)/-I \1 1/"
+		iptables-save -t $1 | grep 'SS_SPEC_CUS_' | sed -e "s/^-A \(zone_lan_forward\)/-I \1 1/" | sed -e "s/^-A \(PREROUTING\)/-I \1 2/"
+		echo 'COMMIT'
+	}
+	extract_rules6() {
+		echo "*$1"
+		ip6tables-save -t $1 | grep 'SS_SPEC_CUS_' | sed -e "s/^-A \(PREROUTING\|zone_lan_forward\)/-I \1 1/"
 		echo 'COMMIT'
 	}
 	cat <<-EOF >>$FWI
@@ -331,6 +337,13 @@ gen_include() {
 		iptables-restore -n <<-EOT
 		$(extract_rules nat)
 		$(extract_rules mangle)
+		$(extract_rules filter)
+		EOT
+
+		ip6tables-save -c | grep -v "SS_SPEC" | ip6tables-restore -c
+		ip6tables-restore -n <<-EOT
+		$(extract_rules6 nat)
+		$(extract_rules6 filter)
 		EOT
 	EOF
 	return 0
