--- feeds/helloworld/chinadns-ng/Makefile	2023-03-04 21:04:35.687983708 +0800
+++ feeds/helloworld/chinadns-ng/Makefile	2023-03-04 21:04:25.907983712 +0800
@@ -5,14 +5,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=chinadns-ng
-PKG_VERSION:=1.0-beta.25
 PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/zfl9/chinadns-ng.git
-PKG_SOURCE_DATE:=2021-05-08
-PKG_SOURCE_VERSION:=14cc6348d67b09cae37d9bce554c89c2c0e0b265
-PKG_MIRROR_HASH:=3b66fc0888d9488e3b8e39df3016d51fae1b43325d292381e94aa3c7d2318282
+PKG_SOURCE_DATE:=%pkg_source_date%
+PKG_SOURCE_VERSION:=%pkg_source_version%
+PKG_MIRROR_HASH:=skip
 
 PKG_LICENSE:=AGPL-3.0-only
 PKG_LICENSE_FILES:=LICENSE
